src/index.ts

```diff
@@ -180,212 +155,208 @@ const getProfileFixture = (p?: Partial<Profile>): Profile =>
+      status: faker.random.arrayElement([
+        NotificationChannelStatusValueEnum.EXPIRED,
+        NotificationChannelStatusValueEnum.FAILED,
+        NotificationChannelStatusValueEnum.THROTTLED,
+        NotificationChannelStatusValueEnum.SENT,
+      ]),
+      channel: faker.random.arrayElement(["EMAIL", "WEBHOOK"]),
+      ...n,
+    }),
+    E.getOrElseW(err => {
+      throw new Error(`Cannot decode notification status payload: ${PR.failure(err).join("\n")}`)
+    })
+  )
 
 const generateServiceFixtures = async (s?: Partial<Service>) => {
   const aService = getServiceFixture(s);
-  const errorOrService = await serviceModel.create(
-    aService,
-    aService.serviceId
-  );
-  const createdService = errorOrService.getOrElseL(() => {
-    throw new Error("cannot create new service");
-  });
+
   // create a new version
-  await serviceModel.update(
-    createdService.id,
-    createdService.serviceId,
-    () => createdService
-  );
+  await pipe(
+    serviceModel.create(
+      aService
+    ),
+    TE.mapLeft(() => {
+      throw new Error("cannot create new service");
+    }),
+    TE.map(service => serviceModel.update(
+      service
+    ))
+  )()
 };
 
-const generateMessageContentFixture = () => {
-  return {
+const generateMessageContentFixture = (): NewMessageContent => {
+  return pipe(NewMessageContent.decode({
     markdown: faker.lorem.words(100),
-    subject: faker.lorem.words(5),
-  } as MessageContent;
+    subject: faker.lorem.words(5)
+  }),
+    E.getOrElseW(() => {
+      throw new Error("Cannot generate message content payload")
+    })
+  )
 };
 
+/**
+  * @deprecated
+*/
+const createOldBlobService = (storageConnectionString: string) => createBlobService(storageConnectionString);
+
 const generateMessageFixtures = async (fiscalCode: FiscalCode) => {
   const aMessage = getMessageFixture({
-    fiscalCode,
+    fiscalCode
   });
-  await messageModel.create(aMessage, aMessage.fiscalCode);
-  await messageModel.storeContentAsBlob(
-    blobService,
-    aMessage.id,
-    generateMessageContentFixture()
-  );
-
-  const aMessageStatus = getMessageStatusFixture({
-    messageId: aMessage.id,
-  });
-  const errorOrMessageStatus = await messageStatusModel.create(
-    aMessageStatus,
-    aMessageStatus.messageId
-  );
 
-  const createdMessageStatus = errorOrMessageStatus.getOrElseL(() => {
-    throw new Error("cannot create new message status");
-  });
-  // create a new version
-  await messageStatusModel.update(
-    createdMessageStatus.id,
-    createdMessageStatus.messageId,
-    () => createdMessageStatus
-  );
-
-  const aNotification = getNotificationFixture({
-    messageId: aMessage.id,
-    fiscalCode: aMessage.fiscalCode,
-  });
-  await notificationModel.create(aNotification, aNotification.messageId);
+  await pipe(
+    messageModel.create(aMessage),
+    TE.chainW(() => {
+      /*
+        NOTICE: io-functions-commons is using a deprecated version for the blob storage.
```

Comment:
what do we need to upgrade the version? Should we track some tickets?

Reply:
io-functions-commons should be updated with the up to date dependencies regarding azure storage. In the code me and @BurnedMarshal decided to utilize the new dependencies in this project while leaving that piece of code for compatibility.